crates/provider/src/builder.rs

```diff
@@ -420,10 +422,13 @@ impl<L, F> ProviderBuilder<L, F, Ethereum> {
     pub fn on_anvil_with_wallet_and_config(
         self,
         f: impl FnOnce(alloy_node_bindings::Anvil) -> alloy_node_bindings::Anvil,
-    ) -> <JoinFill<F, WalletFiller<alloy_network::EthereumWallet>> as ProviderLayer<
-        L::Provider,
-        alloy_transport_http::Http<reqwest::Client>,
-    >>::Provider
+    ) -> Result<
```

Comment:
I'd like to keep this function and not return a result, but we can add `fn try_on_anvil...` for this and on_anvil calls this and unwraps

Reply:
fixed: https://github.com/alloy-rs/alloy/pull/1055/commits/bd30ae136f9ecc4f61354c5a2a17fca832b4ee6f